src/circuit/lookup.rs

```diff
@@ -89,6 +100,10 @@ pub fn lookup3_xy<E: Engine, CS>(
 
     let one = CS::one();
 
+    // After calling `synth`, `x_coeffs` contains the coefficients of the polynomial P
+    // s.t. P(b)=coords(i).x, where b is the binary representation of i
+    // Moving all terms to the lhs except `res_x.get_variable`, we see that the constraint below
+    // corresponds to `P(bits) = res_x.get_variable`
```

Comment:
I followed this locally, and rearranged the equation below as:

`P(000) + aP(001) + bP(010) + baP(011) + cP(100) + caP(101) + cbP(110) + cbaP(111) = P(cba)`

where `bits = [a, b, c]` and `res = P(cba)`.

Reply:
Umm not sure I follow, are you just saying you verified it, or are you suggesting a change?

Reply:
Just that I followed what the comments said, and the above is how I ended up representing things in a way that made what was going on completely clear to me. So I verified it, and maybe it's useful to someone else.